packages/charts/src/Charts/SeverityLine/SeverityLine.js

```diff
@@ -0,0 +1,73 @@
+import './severity-line.scss';
+
+import React, { useEffect, useState } from 'react';
+import { Tooltip, TooltipPosition } from '@patternfly/react-core/dist/js/components/Tooltip/index';
+
+import PropTypes from 'prop-types';
+
+const SeverityLine = ({ title, value, tooltipMessage, config }) => {
+
+    const [ severity, setSeverity ] = useState();
+    const [ isTruncated, setIsTruncated ] = useState(false);
+    const { width, height } = config;
+    const CHART_MAP = {
+        1: '1',
+        2: (width / 3),
+        3: (width * (2 / 3)),
+        4: (width - 1)
+    };
+    const points = [
+        { className: 'horizontal', points: `${1}, ${height / 2} ${width - 1}, ${height / 2}` },
+        { className: 'vertical', points: `${width - 1}, ${0} ${width - 1}, ${height}` },
+        { className: 'vertical', points: `${width * (2 / 3)}, ${0} ${width * (2 / 3)}, ${height}` },
+        { className: 'vertical', points: `${width / 3}, ${0} ${width / 3}, ${height}` },
+        { className: 'vertical', points: `${1}, ${0} ${1}, ${height}` },
+        { className: `dataline-${value}`, points: `${1}, ${height / 2} ${severity}, ${height / 2}` }
+    ];
+    const truncate = (string) => `${string.substring(0, 14)}...`;
+
+    useEffect(() => {
+        value && setSeverity(CHART_MAP[value]);
+        //eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [ value ]);
+
+    useEffect(() => {
+        title.length > 14 && setIsTruncated(true);
```

Comment:
Since we are using this state value only at one place (line 40) you can move this calculation to that line and avoid multiple re-renders (every time `title` changes it fires new render and the value will be calculated ad-hoc.)